packages/sources/coingecko/src/endpoint/crypto.ts

```diff
@@ -101,20 +101,43 @@ const handleBatchedRequest = (
 }
 
 export const execute: ExecuteWithConfig<Config> = async (request, context, config) => {
-  const validator = new Validator(request, inputParameters, {}, { overrides })
+  const validator = new Validator(request, inputParameters, {}, { overrides: internalOverrides })
 
   const endpoint = validator.validated.data.endpoint
   const jobRunID = validator.validated.id
-  const base = validator.overrideSymbol(AdapterName)
+  const base = validator.validated.data.base
   const quote = validator.validated.data.quote
   const coinid = validator.validated.data.coinid
-  let idToSymbol = {}
-  let ids = coinid
-  if (!ids) {
-    const coinIds = await getCoinIds(context, jobRunID)
-    const symbols = Array.isArray(base) ? base : [base]
-    idToSymbol = getSymbolsToIds(symbols, coinIds)
-    ids = Object.keys(idToSymbol).join(',')
+
+  let ids: string
+  let idToSymbol: OverrideToOriginalSymbol = {}
+  if (!coinid) {
+    let overrider: Overrider = {} as Overrider
+    try {
```

Comment:
This feels messy to have here. Can we have the Overrider throw as an `AdapterError`?

Reply:
Also I don't think line 120 is throwing

Reply:
The only reason I did not do it this way is because the AdapterError should include the jobRunID which would require me to pass the JobRunID to the Overrider.